### PR TITLE
Echelle improvements

### PR DESCRIPTION
**Pull request by profxj**
Two changes in this short PR:

1. Allow the usage of `use_2dmodel_mask` for echelle spectrographs.
2. Turns off re-initializing the `reduce_bpm` in the final call to global_sky

The 2nd item was leading to crashes in Echelle data when an order had problems
during object finding.

No new tests or docs, but I am going to run the DevSuite.
And the new functionality has been tested on real world X-Shooter data.

### CODE REVIEW

File: pypeit/find_objects.py [hunk 1]
```diff
@@ -481,7 +482,9 @@ def get_platescale(self, slitord_id=None):
 
 
     def global_skysub(self, skymask=None, update_crmask=True, trim_edg = (0, 0),
-                      previous_sky=None, show_fit=False, show=False, show_objs=False, objs_not_masked=False):
+                      previous_sky=None, show_fit=False, show=False,
```
Comment (debora-pe) [inline pypeit/find_objects.py:486]: `global_skysub` is also defined for the child class for IFU. You should add `reinit_bpm` there as well.

**Review by debora-pe — APPROVED** [review]
All looks good. I only have a small comment. 
Thanks for doing this @profxj!

File: pypeit/find_objects.py (hunk 1 above)
Comment (profxj) [reply pypeit/find_objects.py:486]: done!

be it so noted, @rcooke-ast

**Comment by profxj** [comment]
tests started

**Comment by profxj** [comment]
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  239 passed, 70 warnings in 538.67s (0:08:58) ---
--- PYTEST UNIT TESTS PASSED  133 passed, 171 warnings in 990.98s (0:16:30) ---
--- PYTEST VET TESTS PASSED  58 passed, 68 warnings in 1010.66s (0:16:50) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 231/231 TESTS  ---
Testing Started at 2023-10-05T16:24:11.656047
Testing Completed at 2023-10-06T07:47:17.898775
Total Time: 15:23:06.242728

File: pypeit/find_objects.py [hunk 2]
```diff
@@ -1237,12 +1245,19 @@ def joint_skysub(self, skymask=None, update_crmask=True, trim_edg=(0,0),
...
-                      previous_sky=None, show_fit=False, show=False, show_objs=False, objs_not_masked=False):
+                      previous_sky=None, show_fit=False, show=False, show_objs=False, objs_not_masked=False,
+                      reinit_bpm:bool=True):
         """
         Perform global sky subtraction. This IFU-specific routine ensures that the
         edges of the slits are not trimmed, and performs a spatial and spectral
         correction using the sky spectrum, if requested. See Reduce.global_skysub()
         for parameter definitions.
+
+        Args:
```
Comment (kbwestfall) [inline pypeit/find_objects.py:1260]: Shouldn't necessarily be up to you, but this highlights that this doc-string is missing lots of parameters.  Maybe we point to the documentation in the base class (although that is also missing descriptions...)?

File: pypeit/find_objects.py [hunk 3]
```diff
@@ -1251,7 +1266,7 @@ def global_skysub(self, skymask=None, update_crmask=True, trim_edg=(0,0),
         # Generate a global sky sub for all slits separately
         global_sky_sep = super().global_skysub(skymask=skymask, update_crmask=update_crmask,
                                                trim_edg=trim_edg, show_fit=show_fit, show=show,
```
Comment (kbwestfall) [inline pypeit/find_objects.py:1272]: In looking at the docstring and base class function, I found that the base-class function never uses `trim_edg`.  I.e., in the base-class function (starting at line 484), `trim_edg` is accepted as a keyword argument, but the value passed to `skysub.global_skysub` is from the parset `self.par['reduce']['trim_edge']`.

We don't necessarily need to sort this out in this PR, but, if not, you should raise an issue.

**Review by kbwestfall — APPROVED** [review]
Two things: One minor comment on the docstring; and a possible bug I randomly caught that might never get triggered, but makes the code misleading.  Approving anyway.

File: pypeit/find_objects.py (hunk 2 above)
Comment (profxj) [reply pypeit/find_objects.py:1260]: I've made an effort to update the base class doc string.

File: pypeit/find_objects.py (hunk 3 above)
Comment (profxj) [reply pypeit/find_objects.py:1272]: it appears to be only used in the IFU child.  
Have noted that in the doc.

**Pull request merged**